tests/test_file_contents.py

```diff
@@ -0,0 +1,162 @@
+import pathlib
+import time
+from collections import deque
+from unittest.mock import MagicMock, Mock
+
+import pytest
+import watchfiles
+from watchfiles import Change
+
+from anywidget._file_contents import FileContents
+
+
+def test_file_contents_no_watch(
+    monkeypatch: pytest.MonkeyPatch, tmp_path: pathlib.Path
+):
+    """Test __str__ reads file contents and does not start a thread"""
+    mock = MagicMock()
+    monkeypatch.setattr(watchfiles, "watch", mock)
+
+    CONTENTS = "hello, world"
+    path = tmp_path / "foo.txt"
+    path.write_text(CONTENTS)
+
+    contents = FileContents(path, start_thread=False)
+
+    assert str(contents) == CONTENTS
+    assert contents._background_thread is None
+    mock.assert_not_called
```

Comment:
```suggestion
def test_file_contents_no_watch(tmp_path: pathlib.Path):
    """Test __str__ reads file contents and does not start a thread"""
    CONTENTS = "hello, world"
    path = tmp_path / "foo.txt"
    path.write_text(CONTENTS)

    with patch.object(watchfiles, "watch") as mock:
        contents = FileContents(path, start_thread=False)
        assert str(contents) == CONTENTS
        assert contents._background_thread is None
    mock.assert_not_called()
```

this is the same exact thing... but `with patch...` is a nice way to accomplish this with less code (add `from unittest.mock import patch` at the top).  pytest's monkeypatch fixture uses it under the hood, and in this particular case, it's a little more direct.  but, absolutely the same thing :)

if you like it, it could be used below as well, but also fine to stick with the current code

Reply:
love it. Thanks!